.github/workflows/release.yml

```
@@ -0,0 +1,87 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      candidateName:
+        description: 'Name of the candidate'
+        type: string
+        required: true
+      branchCommit:
+        description: 'Commit to check out. If it is the most recent commit then leave blank.'
+        type: string
+        required: false
+        default: ''
+      cherrypickCommits:
+        description: 'Comma separated commits to cherry pick'
+        type: string
+        required: false
+
+permissions:
+  contents: write
+
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get Monday date
+        id: date
+        run: echo "::set-output name=date::$(date -dmonday +'%Y-%m-%d')"
+      - name: Get releaser identity
+        run: |
+          git config --global user.name '${{github.actor}}'
+          git config --global user.email '${{github.actor}}@users.noreply.github.com'
+      - name: Declare release branch name and tag name
+        id: variables
+        run: |
+          echo "::set-output name=releaseBranchName::release_${CANDIDATE_NAME,,}"
+          echo "::set-output name=tagName::${CANDIDATE_NAME^^}"
+        env:
+          CANDIDATE_NAME: ${{ inputs.candidateName }}
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Create release branch
+        run: |
+          if [[ -z "${{ inputs.branchCommit }}" ]]; then
+            git checkout -b $RELEASE_BRANCH_NAME
+          else
+            git checkout -b $RELEASE_BRANCH_NAME $BRANCH_COMMIT
+          fi
```

```suggestion
          git checkout -b $RELEASE_BRANCH_NAME $BRANCH_COMMIT
```

I think this can be simplified since in the case where we enter the if, $BRANCH_COMMIT is '' anyways